src/engine/irc.cpp

```diff
@@ -360,19 +360,31 @@ ICOMMAND(0, ircbind, "ss", (const char *name, const char *s), {
 ICOMMAND(0, ircpass, "ss", (const char *name, const char *s), {
     ircnet *n = ircfind(name);
     if(!n) { conoutf("no such ircnet: %s", name); return; }
-    if(!s || !*s) { ircprintf(n, 4, NULL, "current password is: %s", n->passkey && *n->passkey ? "<set>" : "<not set>"); return; }
+    if(!s || !*s)
```

Comment:
While we're at it, can you directly fix these pointer calculations to be more explicit?

```suggestion
    if(s == 0 || *s != '\0')
```

Reply:
sure ! do you want me to find all these occurences everywhere in the code and apply the same ?

Reply:
Can do it in this PR, if you limit it to the methods you touch anyway.

Reply:
using `s == 0` will trigger a new warning: `zero-as-nullptr`.

dctremblay: we clean the code step after step, keeping commits as limited as possible eases the reviews.

Reply:
`*s != '\0'` is not the same as `!*s`. It should be `*s == '\0'`

Reply:
Yep, exactly.